src/kernels/jupyter/session/backingFileCreator.base.ts

```diff
@@ -17,6 +17,13 @@ function getRemoteIPynbSuffix(): string {
     return `${jvscIdentifier}${uuid()}`;
 }
 
+export function generateBackingIPyNbFileName(resource: Resource) {
```

Comment:
Generic fucntion so that we can have a meaningful name, if we fail to create the backing file. Previously it used to fallback to `<uuid>.ipynb`, and that was ugly.
E.g. point to a jupyter server that doesn't support creation of check points

Reply:
Yes agree, if you see the code again, the `suffix` does contain a uuid, hence it is unique.